src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SnakeRoadCheck.java

```diff
@@ -160,6 +161,10 @@ private boolean networkWalkQualifiesAsSnakeRoad(final SnakeRoadNetworkWalk walk)
      */
     private void walkNetwork(final Edge current, final SnakeRoadNetworkWalk walk)
     {
+        final Set<Edge> directConnections = walk.getConnectedMasterEdgeOfTheSameWay(current);
```

Comment:
to avoid reprocessing, we can just say ```final Set<Edge> directConnections = walk.getDirectConnections()``` here. Might need to change Set to a LinkedList doing this

Reply:
Done